Compiler/src/Compiler.cpp

```diff
@@ -1123,6 +1130,10 @@ struct Compiler
                 std::swap(left, right);
         }
 
+        // disable fast path for vectors because supporting it would require a new opcode
+        if (isConstantVector(right))
```

Comment:
```suggestion
        if (operandIsConstant && isConstantVector(right))
```
Saves a constant lookup if we already know we won't find it.